extends Node2D
class_name SpeedComponent

## –ú–∞–∫—Å–∏–º–∞–ª—å–Ω–∞—è —Å–∫–æ—Ä–æ—Å—Ç—å —é–Ω–∏—Ç–∞. –ï—Å–ª–∏ –º–µ–Ω—å—à–µ 0 –ø—Ä–∏ —Ç–∏–∫–µ, —Ç–æ –≤–æ–∑–≤—Ä–∞—â–∞–µ—Ç—Å—è –∫ —ç—Ç–æ–π —Å–∫–æ—Ä–æ—Å—Ç–∏
@export var max_speed := 3
## –õ–æ–∫–∞–ª—å–Ω–∞—è —Å–∫–æ—Ä–æ—Å—Ç—å —é–Ω–∏—Ç–∞
@export var speed := 2:
	set(new_speed):
		speed = new_speed
		update_hint()
## –•–ó —á—Ç–æ —ç—Ç–æ. –ú–± –æ—Ç–≤–µ—Ç—á–∞–µ—Ç –∑–∞ —Ç–µ–∫—É—â–∏–π —Ö–æ–¥?
@export var isMyTurn := false

func _ready():
	speed = max_speed
	
var format_hint = "üëü: {speed}({max_speed})"
	
func update_hint():
	$SpeedHint.text = format_hint.format({"speed": speed, "max_speed": max_speed})

## —Ñ—É–Ω–∫—Ü–∏—è —Ç–∏–∫–∞ —Å–∫–æ—Ä–æ—Å—Ç–∏ —é–Ω–∏—Ç–∞. –ï—Å–ª–∏ —Å–∫–æ—Ä–æ—Å—Ç—å –¥–æ—Ö–æ–¥–∏—Ç –¥–æ 0, —Ç–æ —Ö–æ–¥–∏—Ç —é–Ω–∏—Ç –∏ –Ω—É–∂–Ω–æ –æ—Å—Ç–∞–Ω–æ–≤–∏—Ç—å –¥—Ä—É–≥–∏–µ —Ç–∏–∫–∏
func tick():
	if speed < 1:
		speed = max_speed
	else:
		speed -= 1
	print('tick', speed)
